client/routes/workflow/components/workflow-graph/index.vue

```diff
@@ -0,0 +1,224 @@
+<script>
+// Copyright (c) 2020-2021 Uber Technologies Inc.
+//
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import omit from 'lodash-es/omit';
+import Graph from './components/graph.vue';
+import GraphLegend from './components/graph-legend.vue';
+import { GRAPH_SHOW_DELAY } from './constants';
+
+export default {
+  name: 'workflow-graph',
+  props: ['events', 'isWorkflowRunning', 'selectedEventId'],
+  components: {
+    Graph,
+    GraphLegend,
+  },
+  data() {
+    return {
+      isGraphLoading: true,
+      forceRefresh: true,
+      eventsSnapShot: [],
+    };
+  },
+  methods: {
+    delayedShow() {
+      setTimeout(() => {
+        this.isGraphLoading = false;
+      }, GRAPH_SHOW_DELAY);
+    },
+    updateRoute(route) {
+      const params = route.workflowId
+        ? { workflowId: route.workflowId, runId: route.runId }
+        : { runId: route };
+
+      this.$router.replace({
+        params,
+        query: omit(this.$route.query, 'eventId'),
+      });
+    },
+    reloadWorkflow() {
+      this.eventsSnapShot = this.events;
+      this.isGraphLoading = true;
+      this.forceRefresh = !this.forceRefresh;
```

Comment:
This looks suspicious. Why toggling is needed here?

Reply:
I think whats going on here is its forcing the graph to refresh by toggling the boolean flag.

Reading this article shows two accepted ways of forcing a re-render:
https://michaelnthiessen.com/force-re-render/

This shows one of the methods which is using the key method. In the example it uses an integer to increment to larger values each time, here we toggle between true and false. We could change this to increment the counter like the example. Alternatively we could use `this.$forceUpdate();` which will force the entire component to re-render (not just the graph).

Let me know your thoughts and I will go with that approach.

Thanks!

Reply:
Also the reason why we are forcing a refresh is I think the graph component is not reactive as it is built using a jQuery library.

Reply:
Ahh, ok understood. No need to change it, this is elegant enough.
You could add a comment with the link explaining the technique though. Could be helpful for those not familiar with it.

Reply:
Comment added. Thanks.